jerry-core/api/jerry.c

```diff
@@ -5110,6 +5110,100 @@ jerry_get_container_type (const jerry_value_t value) /**< the container object *
 } /* jerry_get_container_type */
 
+/**
+ * Perform container operation on the given operands (add, get, set, has, delete, size, clear).
+ *
+ * @return error - if argument has an error flag or operation is unsuccessful or unsupported
+ *         result of the container operation - otherwise
+ */
+jerry_value_t
+jerry_container_operation (jerry_container_operation_t operation, /**< container operation */
+                           jerry_value_t container, /**< container */
+                           jerry_value_t *arguments, /**< list of arguments */
+                           uint32_t arguments_number) /**< number of arguments */
+{
+  jerry_assert_api_available ();
+
+  if ((arguments_number >= 1 && ecma_is_value_error_reference (arguments[0]))
+      || (arguments_number >= 2 && ecma_is_value_error_reference (arguments[1])))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (error_value_msg_p)));
+  }
+
+  if (ecma_is_value_object (container))
+  {
+#if ENABLED (JERRY_BUILTIN_CONTAINER)
+    ecma_object_t *obj_p = ecma_get_object_from_value (container);
+
+    if (ecma_get_object_type (obj_p) == ECMA_OBJECT_TYPE_CLASS)
+    {
+      uint16_t type = ((ecma_extended_object_t *) obj_p)->u.class_prop.class_id;
+      ecma_extended_object_t *map_object_p = ecma_op_container_get_object (container, type);
+
+      if (map_object_p == NULL)
+      {
+        return ecma_create_error_reference_from_context ();
+      }
+
+      jerry_value_t result;
+
+      switch (operation)
+      {
+#if ENABLED (JERRY_BUILTIN_SET) || ENABLED (JERRY_BUILTIN_WEAKSET)
+        case JERRY_CONTAINER_OP_ADD:
+        {
+          result = jerry_return (ecma_op_container_set (map_object_p, arguments[0], arguments[0], type));
+        }
+#endif /* ENABLED (JERRY_BUILTIN_SET) || ENABLED (JERRY_BUILTIN_WEAKSET) */
+#if ENABLED (JERRY_BUILTIN_MAP) || ENABLED (JERRY_BUILTIN_WEAKMAP)
+        case JERRY_CONTAINER_OP_GET:
+        {
+          result = jerry_return (ecma_op_container_get (map_object_p, arguments[0], type));
+        }
+        case JERRY_CONTAINER_OP_SET:
+        {
+          result = jerry_return (ecma_op_container_set (map_object_p, arguments[0], arguments[1], type));
+        }
+#endif /* ENABLED (JERRY_BUILTIN_MAP) || ENABLED (JERRY_BUILTIN_WEAKMAP) */
+#if ENABLED (JERRY_BUILTIN_SET) || ENABLED (JERRY_BUILTIN_WEAKSET) \
+        || ENABLED (JERRY_BUILTIN_MAP) || ENABLED (JERRY_BUILTIN_WEAKMAP)
+        case JERRY_CONTAINER_OP_HAS:
+        {
+          result = jerry_return (ecma_op_container_has (map_object_p, arguments[0], type));
+        }
+        case JERRY_CONTAINER_OP_DELETE:
+        {
+          if (type == LIT_MAGIC_STRING_WEAKMAP_UL || type  == LIT_MAGIC_STRING_WEAKSET_UL)
+          {
+            result = jerry_return (ecma_op_container_delete (map_object_p, arguments[0], type));
+          }
+          result = jerry_return (ecma_op_container_delete (map_object_p, arguments[0], type));
+        }
```

Comment:
What is the difference between the two cases (the on inside the if and outside)?